src/Juvix/Compiler/Core/Extra/Utils.hs

```diff
@@ -58,6 +63,27 @@ filterOutTypeSynonyms tab = pruneInfoTable tab'
     tab' = tab {_infoIdentifiers = idents'}
     idents' = HashMap.filter (\ii -> not (isTypeConstr tab (ii ^. identifierType))) (tab ^. infoIdentifiers)
 
+isType :: Node -> Bool
+isType = \case
+  NPi {} -> True
+  NUniv {} -> True
+  NPrim {} -> True
+  NTyp {} -> True
+  NDyn {} -> True
+  NBot (Bottom i) -> isType (Info.getInfoType i)
```

Comment:
Change this to `False`. This is not entirely correct, but there really isn't a reliable way to check if the type of Bottom is a type constructor (and that's not even what the code here tries to check). There is no type info available, unless you always store it when creating Bottom nodes (otherwise it cannot be computed in general, because where would you get it from?).

Reply:
done in #e8e41f41c